pkg/agent/datamodel/types.go

```diff
@@ -662,7 +662,8 @@ type AgentPoolProfile struct {
 	MessageOfTheDay       string               `json:"messageOfTheDay,omitempty"`
 	// This is a new property and all old agent pools do no have this field. We need to keep the default
 	// behavior to reboot Windows node when it is nil
-	NotRebootWindowsNode *bool `json:"notRebootWindowsNode,omitempty"`
+	NotRebootWindowsNode *bool                    `json:"notRebootWindowsNode,omitempty"`
+	WindowsProfile       *AgentPoolWindowsProfile `json:"agentPoolWindowsProfile,omitempty"`
```

Comment:
use `agentPoolWindowsProfile` to differentiate from the property in cluster level?

Reply:
I think that we can also use agentPoolWindowsProfile in API. If not, we can use `agentPoolWindowsProfile` in code but expose windowsProfile in API

Reply:
Updated to AgentPoolWindowsProfile.